test/com/ecfeed/core/generators/algorithms/ScoreBasedAlgorithmTest.java

```diff
@@ -0,0 +1,32 @@
+package com.ecfeed.core.generators.algorithms;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.ecfeed.core.evaluator.HomebrewConstraintEvaluator;
+import com.ecfeed.core.generators.api.GeneratorException;
+
+public class ScoreBasedAlgorithmTest {
+	
+	@Test
+	public void testproduceTestSuite() throws GeneratorException {
+		
+		List<List<String>> input = Arrays.asList(Arrays.asList("a1", "a2"), Arrays.asList("b1", "b2"),
+				Arrays.asList("c1", "c2"), Arrays.asList("d1", "d2"));
+		NwiseScoreEvaluatorMock<String> fScores = new NwiseScoreEvaluatorMock<>(input, new HomebrewConstraintEvaluator(input), 3);
```

Comment:
HomebrewConstraintEvaluator must be legacy from Przemek. I must have not notice that name when I reviewed his code. Can you go through the sources and rename it to something more adequate? Maybe 'SatSolverConstraintEvaluator'